world/economy.py

```diff
@@ -43,10 +43,10 @@ def coin_to_value(coins):
     Args:
         coins (int, dict): dict of coin names and values; assumes CC if an int
     """
-    if not isinstance(coins, (int, dict, _SaverDict)) or coins is None:
-        raise TypeError("'coins' must be a dict of 'coin': count pairs.")
     if coins is None:
         return None
+    if not isinstance(coins, (int, dict, _SaverDict)):
```

Comment:
Note that `_SaverDict` is a private implementation detail (the initial `_` generally denote privacy - well as private as things get in Python). While unlikely to change it's a bit of bad form to rely on it for checking (I know the check was not introduced in this contrib but it's just where I happened to notice it). More Pythonic would be to not do a typecheck here but assume it's a dict and catch and handle the traceback if it's not.